controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -332,33 +332,28 @@ else if ( ! application.deploying().isPresent() && ! zone.environment().isManual
                                                        " the current version " + existingDeployment.version());
             }
 
-            Optional<Rotation> rotation;
-            try (RotationLock rotationLock = rotationRepository.lock()) {
-                rotation = getRotation(application, zone, rotationLock);
-                if (rotation.isPresent()) {
-                    application = application.with(rotation.get().id());
+            Set<String> rotationNames = new HashSet<>();
+            Set<String> cnames = new HashSet<>();
+            if (zone.environment() == Environment.prod && application.deploymentSpec().globalServiceId().isPresent()) {
+                try (RotationLock rotationLock = rotationRepository.lock()) {
+                    Rotation rotation = rotationRepository.getRotation(application, rotationLock);
+                    application = application.with(rotation.id());
                     store(application); // store assigned rotation even if deployment fails
-                    registerRotationInDns(rotation.get(), application.rotation().get().dnsName());
+
+                    rotationNames.add(rotation.id().asString());
+                    application.rotation().ifPresent(applicationRotation -> {
+                        registerRotationInDns(rotation, applicationRotation.dnsName());
+                        cnames.add(applicationRotation.dnsName());
+                        registerRotationInDns(rotation, applicationRotation.secureDnsName());
+                        cnames.add(applicationRotation.secureDnsName());
```

Comment:
Adding also the secureDnsName, which is `https://service--(instance, but not really--)application--tenant` to both DNS and to the cnames sent to the zone, and finally to the routing config. I hope they don't end up anywhere else ... !

Reply:
Keep in mind these endpoint (aliases) are returned to the controller, and to the `DeploymentTester`, from the routing layer, as among the endpoints for a given application, and are then made visible in, e.g., the dashboard. 
Since this endpoint names may be used in automated stuff, I don't want to make them visible yet. I will hide them on the routing node, i.e., not include them in the endpoint query response. This is in hosted, obviously.